pallets/ethereum-transaction/src/lib.rs

```diff
@@ -129,7 +151,21 @@ pub mod pallet {
 					.map_or(Some(read_weight), |weight| {
 						Some(weight.saturating_add(read_weight))
 					}),
-			})
+			})?;
+
+			// NOTE: The Ethereuem side of things never returns a DispatchError
+			//       if the execution failed. But we can check that manually by
+			//       querying the `Pending` storage of the pallet-ethereum.
+			let pending = pallet_ethereum::Pending::<T>::get();
+			let last = pending.last().ok_or(DispatchError::Other(
+				"Ethereuem not adding pending storage. Unexpected.",
+			))?;
+
+			if Pallet::<T>::valid_code(&last.2) {
```

Comment:
Ultra nit: Always in favor of destructuring tuples to be aware of breaking changes to number of tuple elements.
```suggestion
			let (_, last_receipt, _) = pending.last().ok_or(DispatchError::Other(
				"Ethereuem not adding pending storage. Unexpected.",
			))?;

			if Pallet::<T>::valid_code(&last_receipt) {
```

Reply:
Done.